apps-rendering/src/components/TableOfContent/index.tsx

```diff
@@ -0,0 +1,125 @@
+import { css, SerializedStyles } from '@emotion/react';
+import { text } from '@guardian/common-rendering/src/editorialPalette/text';
+import { ArticleFormat } from '@guardian/libs';
+import {
+	line,
+	neutral,
+	remSpace,
+	textSans,
+} from '@guardian/source-foundations';
+import {
+	SvgChevronDownSingle,
+	SvgChevronUpSingle,
+} from '@guardian/source-react-components';
+import Anchor from 'components/Anchor';
+import ListItem from 'components/ListItem';
+import OrderedList from 'components/OrderedList';
+import { Outline } from 'outline';
+import { FC } from 'react';
+import { darkModeCss } from 'styles';
+
+interface Props {
+	format: ArticleFormat;
+	outline: Outline;
+}
+
+const anchorStyles = (format: ArticleFormat): SerializedStyles => css`
+	color: ${text.paragraph(format)};
+	border-bottom: none;
+	:hover {
+		border-bottom: 0.0625rem solid ${neutral[86]};
+	}
+`;
+
+const listStyles: SerializedStyles = css`
+	> li::before {
+		content: none;
+	}
+
+	margin: 0;
+
+	> li {
+		padding-left: ${remSpace[3]};
+	}
+`;
+
+const listItemStyles: SerializedStyles = css`
+	${textSans.xsmall({ fontWeight: 'bold', lineHeight: 'regular' })}
+	border-bottom: ${line.primary} 1px solid;
+	padding-top: ${remSpace[2]};
+`;
+
+const detailsStyles: SerializedStyles = css`
+	margin-bottom: 1.25rem;
+	&:not([open]) .is-on,
+	&[open] .is-off {
+		display: none;
+	}
+	summary::-webkit-details-marker {
+		display: none;
+	}
+`;
+
+const summaryStyles: SerializedStyles = css`
+	cursor: pointer;
+	position: relative;
+	list-style: none;
+	align-items: center;
+	padding-left: ${remSpace[3]};
+	padding-top: 0.44rem;
+	padding-bottom: 0.375rem;
+	border-bottom: ${line.primary} 1px solid;
+	border-top: ${line.primary} 1px solid;
+
+	path {
+		fill: ${neutral[46]};
+	}
+	svg {
+		height: 2rem;
+	}
+`;
+
+const titleStyle: SerializedStyles = css`
+	${textSans.xsmall({ lineHeight: 'regular' })}
+	color: ${neutral[46]};
+	${darkModeCss`
+		color: ${neutral[86]};
+	`}
+`;
+
+const arrowPosition: SerializedStyles = css`
+	position: absolute;
+	right: ${remSpace[1]};
+	top: 0;
+`;
+
+const TableOfContent: FC<Props> = ({ format, outline }) => {
+	return (
+		<details open={outline.length < 5} css={detailsStyles}>
+			<summary css={summaryStyles}>
+				<h2 css={titleStyle}>Jump to...</h2>
+				<span className="is-off" css={arrowPosition}>
+					<SvgChevronDownSingle size="xsmall" />
+				</span>
+				<span className="is-on" css={arrowPosition}>
+					<SvgChevronUpSingle size="xsmall" />
+				</span>
+			</summary>
+			<OrderedList className={listStyles}>
+				{outline.map((ol) => (
+					<ListItem className={listItemStyles}>
+						<Anchor
+							format={format}
+							href={'#' + ol.id}
+							className={anchorStyles(format)}
+						>
+							{ol.doc.textContent}
```

Comment:
I think it's possible we want to run this through a rendering step that's similar to what we use for the `h2`s in the body? If we just use the `textContent` we'll lose any emphasis, superscript, subscript etc.?

It may not necessarily be an identical rendering step, however. For example, in the body we allow anchors in the `h2`, whereas if we allow those here they'll conflict with the enclosing anchor.

Reply:
Thanks @JamieB-gu for your suggestion, it's a valid point. I did add a new `TocTextElement` within the table of content which is now rendering the node considering the nested sub elements. Could you check if it looks correct or I couldv'e done it in a better way? 

These changes can be found in this commit [render toc contents with their existing nested html elemenst](https://github.com/guardian/dotcom-rendering/pull/5841/commits/0e8de40ff2788bb943a60e9db54fabb4ef91dc10)